### PR TITLE
feat(auth): use schema-driven account form validation

### LINKED ISSUE #73: feat(auth): add accessible schema-based form validation
## Outcome
Give login and registration consistent, accessible inline validation using React Hook Form + Zod with shadcn field/error components, following https://ui.shadcn.com/docs/forms/react-hook-form.

## Scope
Replace the current browser-tooltip-led form experience with clear inline errors. React Hook Form manages state/submission; Zod defines schemas; shadcn provides presentation. No Formik dependency. Preserve server-side validation and existing session safeguards.

## Acceptance criteria
- [ ] Login and registration use the shared approach with distinct schemas matching backend requirements; preserve login versus registration password limits and never trim passwords.
- [ ] Empty/invalid submission shows inline field errors and focuses the first invalid field, without page reload, navigation or API requests. Errors revalidate as corrected after submission.
- [ ] Labels, aria-invalid, error descriptions and pending states work with keyboard and screen readers. Avoid native tooltip interference in hydrated forms while preserving semantic input attributes and pre-hydration disabled/POST safeguards.
- [ ] Preserve valid sign-in, registration confirmation, generic server errors, timeout/retry behavior and duplicate-submit prevention. Never log credentials or display them in diagnostic toasts.
- [ ] Document the reusable form pattern for future forms; backend remains authoritative.
- [ ] Add schema/component tests and Playwright desktop/mobile coverage for empty fields, malformed email, mode-specific password boundaries, correction, server failures, successful account journey and delayed hydration; run accessibility checks.
- [ ] Self-review, clean current-head Codex review and required CI pass before merge; verify development and production behavior after delivery and update relevant docs/board.

## Delivery order — 11 September
First planned implementation tomorrow, in a dedicated issue-linked PR. Then #71 (cart API), followed by #72 (cart storefront, dependent on #71). One active implementation at a time; this is an ordered queue, not a promise to finish all three in a day. #67 PR-template cleanup and other maintenance remain separate Backlog work.

## Status
Planning only. Implementation has not started.

## Design-system reference
Follow https://github.com/youneshenniwrites/fastapi-next-ecommerce/wiki/Design-system (published in #74). Update the planned forms section to verified implementation when this ticket is delivered.

### PR DESCRIPTION
**Pull request by youneshenniwrites**
## Summary
Login and registration now show accessible inline validation using React Hook Form, Zod and shadcn Field components. Library-managed validation and focus replace browser tooltip feedback while preserving session and submission safeguards.

## Issue
Closes #73.

## Before / After
| Before | After |
| --- | --- |
| Browser required-field tooltip | Inline field errors with associated descriptions and first-invalid focus |
| Native constraint feedback | Zod schemas, React Hook Form submit/change validation and shadcn presentation |

## Acceptance criteria
- [x] Separate login/registration constraints; email trimming and unchanged passwords.
- [x] Empty/invalid submissions cause no navigation or API calls.
- [x] Pending, failure, account journey and delayed hydration safeguards preserved.
- [x] Reusable form guidance documented; Wiki update follows delivery.

## Testing
Local production build, lint, formatting and 55 unit tests passed. Playwright: 37 passed, two expected desktop skips, covering desktop/mobile account journeys, error correction, accessibility, delayed hydration, session privacy and catalog regressions.

## Review
Self-review completed across the implementation, generated components, dependencies and tests. External Codex review and CI pending.

## Deployment / Compatibility
No API or database changes. Backend validation remains authoritative. Verify hosted forms after merge; publish the Wiki implementation update then.

## Summary by CodeRabbit

- **New Features**
  - Added client-side validation for account forms, including email and password requirements.
  - Added inline error messages, invalid-field accessibility states, and automatic focus on the first invalid field.
  - Added reusable form field, label, and separator components.

- **Tests**
  - Expanded validation coverage for required fields, email formats, password limits, error clearing, and accessibility.

### CODE REVIEW

**Comment by youneshenniwrites** [comment]
@codex review

File: frontend/design-system.md [hunk 1]
```diff
@@ -51,3 +51,14 @@ Run lint, formatting, types, unit tests, build and Playwright as described in th
...
+## Account form validation
+
+Account forms use React Hook Form with a Zod resolver and shadcn Field, FieldLabel
+and FieldError. Schemas live in src/lib/account-validation.ts. Validate on submit,
+then on change; associate inline messages with inputs and focus the first invalid
+field. Hydrated forms suppress native tooltips with noValidate. Preserve required
+attributes, pre-hydration disabled controls and POST fallback. Login and registration
+use different password limits; never trim or log passwords. API validation remains
+authoritative. Test invalid submissions for zero navigation and zero API calls.
+See the [design-system Wiki](https://github.com/youneshenniwrites/fastapi-next-ecommerce/wiki/Design-system).
```
[the remark this answers is not in the document]
Comment (youneshenniwrites) [reply frontend/design-system.md:64]: Addressed in Wiki commit cd3167c9f9749e3fd7bc41e3fd6f6292384cab57. The Design-system page now documents useForm, zodResolver and shadcn Field components implemented in this PR, with merge/rollout explicitly pending rather than claiming production has changed. Updated the component inventory too. Verified the published remote revision and git diff --check. Application evidence remains 55 unit tests and 37 browser tests passing, with two expected desktop skips. https://github.com/youneshenniwrites/fastapi-next-ecommerce/wiki/Design-system
